packages/react-components/src/components/RichTextEditor/Plugins/UpdateContentPlugin.tsx

```diff
@@ -70,20 +71,14 @@ export class UpdateContentPlugin implements EditorPlugin {
 
       case PluginEventType.ContentChanged:
         if (
-          event.source.toLowerCase() === 'cut' ||
-          (event.source.toLowerCase() === 'keyboard' && (event.data === Keys.BACKSPACE || event.data === Keys.DELETE))
+          event.source === ChangeSource.Cut ||
+          (event.source === ChangeSource.Keyboard && (event.data === Keys.BACKSPACE || event.data === Keys.DELETE))
         ) {
-          imageSrcArray = [];
-          event.contentModel?.blocks.map((block) => {
-            if (block.blockType === 'Paragraph') {
-              const segments = block.segments;
-              segments.map((segment) => {
-                if (segment.segmentType === 'Image') {
-                  imageSrcArray?.push(segment.src);
-                }
-              });
-            }
-          });
+          imageSrcArray = getImagesSourcesList(event.contentModel);
+        } else if (event.source === ChangeSource.SetContent) {
```

Comment:
I know that undo and redo both have a ChangeSource of SetContent. I wonder whether there's other action that also under SetContent? We need to make sure not to accidentally call onRemove on unexpected actions.

Reply:
This functionality only go over the images so if the image exists for content model blocks. We don't delete anything here
Also, it looks like the SetContent event is sent for undo/redo only
https://github.com/search?q=repo%3Amicrosoft%2Froosterjs+ChangeSource.SetContent&type=code